### PR TITLE
hugin-app: Switch to wxWidgets version 3.2

### PR DESCRIPTION
**Pull request by FlyingSamson**
#### Description
Previously if the system uses the dark theme, this resulted in white font on white buttons. By moving from wxWidgets 3.0 to 3.2 the Hugin UI correctly adopts to the used theme.

For reference: Darkmode support was added to wxWidget in version 3.1.3 according to the [changelog](https://raw.githubusercontent.com/wxWidgets/wxWidgets/v3.1.3/docs/changes.txt)

If wxWidgets 3.0 should be required on older macOS versions, one might consider adding a `if`-`else` block to select version 3.0 on those systems, but I think on modern systems the default should be to use version 3.2 for seamless system integration.

Here some screenshots showing the GUI before/after the change with dark mode enabled and disabled

Before change with dark mode enabled:
Screenshot 2023-10-23 at 19 34 57

After change with dark mode enabled:
Screenshot 2023-10-23 at 19 30 17

Before change with dark mode disabled:
Screenshot 2023-10-23 at 19 35 12

After change with dark mode disabled:
Screenshot 2023-10-23 at 19 30 49

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 13.6 22G120 arm64
Xcode 15.0.1 15A507

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by mascguy** [comment]
Thanks, this will be a nice enhancement.

I need to verify whether 3.2 is feasible for earlier macOS releases though. So let's keep this PR open, until that's done.

Will report back when I can find some time, over the coming days.

**Comment by barracuda156** [comment]
@FlyingSamson Is the app broken on systems which do not currently support wxWidgets 3.2? If not, then a newer version should be rather set conditionally.

**Comment by mascguy** [comment]
> @mascguy wxWidgets 3.2 will certainly not build on any powerpc system at the moment, as well as, I believe, 10.4–10.5 Intel. Of course, if the port is known to be broken on all those systems anyway, unrelated to wxWidgets, then it is inconsequential.

Yes, per my previous comments, I'll ensure we determine where we need to fall back.

**Comment by FlyingSamson** [comment]
Maybe it suffices to restrict the usage of wxWidgets 3.2 to the macOS version that actually support dark mode?
This would mean macOS Mojave (10.14) or later. For OSs before that I don't think much is gained by updating to version 3.2.

The "port health" of wxwidgets 3.2 seems to be good for this range of macOS versions.

**Comment by pmetzger** [comment]
@FlyingSamson I don't think we should delay this any longer. Can you fix the build failure on the MacOS 13 runner?

**Comment by barracuda156** [comment]
@pmetzger @FlyingSamson Just make sure please that the switch is done for systems which actually benefit from it and does not cause a breakage for older ones. PPC may not be working anyway (and I won’t check now), but there is no point in breaking older Intel.

**Comment by pmetzger** [comment]
@barracuda156 If you have access to such machines you can always submit patches to fix them. It's unreasonable to expect normal people to have access to boxes like that to test, or to take them heavily into consideration when trying to maintain a port. It's up to the people who want to keep obsolete hardware running to do that, not up to people maintaining ordinary ports.

**Comment by barracuda156** [comment]
@pmetzger See above:

> Maybe it suffices to restrict the usage of wxWidgets 3.2 to the macOS version that actually support dark mode?
> This would mean macOS Mojave (10.14) or later.

So make this change for Mojave and later, that’s it. It is unneeded for anything earlier.

**Comment by pmetzger** [comment]
Just to be clear: if something passes the CI, works on recent MacOS releases, and works on the maintainer's computer, then it is up to other people to submit patches for old machines that do not harm use on recent machines. It is not up to people who are working on recent machines to make sure it does not harm use on old machines. It's fine for people to keep the old stuff running to the extent that it isn't a burden on people doing work on new hardware and operating systems.

**Comment by barracuda156** [comment]
@pmetzger There is nothing here about maintaining my PowerPC or any PowerPC at all. We know that `wxWidgets-3.2` does not build for a number of systems. It is just wrong to force those systems use `wxWidgets-3.2` instead of `wxWidgets-3.0`. I can’t see what is controversial here.

**Comment by FlyingSamson** [comment]
I see both your points.

The current port file already contains the distinction in system >= 10.14 where wxWidget v3.2 is used and older systems where wxWidgets v3.0 continues to be used, because dark mode is only supported since 10.14.

Of course, if at one point another good reason arises for using wxWidgets v3.2 also on older system, one could rethink this distinction. And of course if one day the wxWidgets v3.0 port should be dropped entirely this would also need to be revisited. Until then I think this might be an easy way to optimally support old and new systems alike.

Regarding the failing pipeline on macOS 13: I don't think that has anything to do with the changes in this pull request as it is failing already in the "Bootstrap MacPorts" step after trying for about 50 minutes. But I think I don't have sufficient privileges to restart that pipeline.

File: graphics/hugin-app/Portfile
Comment (reneeotten) [inline graphics/hugin-app/Portfile:37]: this check is not correct. It should first check whether it's being used on macOS and then `os.major` refers to the Darwin version ([see this link](https://trac.macports.org/wiki/XcodeVersionInfo#macOSVersionMap)).

So it should read: ```if {${os.platform} eq "darwin" && ${os.major} >= 18} {```.

Once you make that change we'll merge.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: graphics/hugin-app/Portfile
Comment (FlyingSamson) [reply graphics/hugin-app/Portfile]: Thanks for the hint. I now adapted the portfile accordingly.

**Comment by reneeotten** [comment]
thanks for bearing with us @FlyingSamson ; getting this finalized took way to much time from our side!

**Comment by FlyingSamson** [comment]
No worries!

**Pull request merged**
